lib/BufferUtil.fallback.js

```diff
@@ -16,36 +16,14 @@ exports.BufferUtil = {
     }
   },
   mask: function(source, mask, output, offset, length) {
-    var maskNum = mask.readUInt32LE(0, true);
-    var i = 0;
-    for (; i < length - 3; i += 4) {
-      var num = maskNum ^ source.readUInt32LE(i, true);
-      if (num < 0) num = 4294967296 + num;
-      output.writeUInt32LE(num, offset + i, true);
+    for (var i = 0; i < length; i++) {
+      output[offset + i] = source[i] ^ mask[i & 3];
     }
-    /* eslint-disable no-fallthrough */
-    switch (length % 4) {
-    case 3: output[offset + i + 2] = source[i + 2] ^ mask[2];
-    case 2: output[offset + i + 1] = source[i + 1] ^ mask[1];
-    case 1: output[offset + i] = source[i] ^ mask[0];
-    }
-    /* eslint-enable no-fallthrough */
   },
   unmask: function(data, mask) {
-    var maskNum = mask.readUInt32LE(0, true);
     var length = data.length;
```

Comment:
@Nibbler999 can you use `data.length` directly in the for loop? This variable is not very useful.